governance/src/lib.rs

```diff
@@ -29,15 +29,33 @@ pub struct Governance<N: GossipNetwork, S: Storage> {
 
 impl<N: GossipNetwork, S: Storage> Governance<N, S> {
     pub async fn create(_dms: DMS<N, S>, _height: BlockHeight) -> Result<(), Error> {
-        unimplemented!()
+        Ok(())
     }
 
-    pub async fn open(_dms: DMS<N, S>) -> Result<Self, Error> {
-        unimplemented!()
+    pub async fn open(dms: DMS<N, S>) -> Result<Self, Error> {
+        Ok(Self { dms })
     }
 
-    pub async fn read(&self) -> Result<GovernanceState, Error> {
-        unimplemented!()
+    pub async fn read(&self) -> Result<GovernanceStatus, Error> {
+        let messages = self.dms.read_messages().await?;
+        let votes = messages
+            .iter()
+            .map(|message| {
+                (
+                    message.data().to_string().to_hash256(),
```

Comment:
We need `agenda_hash`, not the hash of the data.

Reply:
fixed